From e18edc2664051af8024169f748f321a260e0fb10 Mon Sep 17 00:00:00 2001
From: Johnny Mnemonic <jm@machine-hall.org>
Date: Tue, 25 Mar 2025 10:21:52 +0100
Subject: [PATCH 65/90] Revert "vmlinux.lds.h: Remove entry to place init_task
 onto init_stack"

ia64 still selects ARCH_TASK_STRUCT_ON_STACK, so we keep this.

Related:
* [0eb5085](https://github.com/torvalds/linux/commit/0eb5085c38749f2a91e5bd8cbebb1ebf3398343c)
* [79a7beb](https://github.com/johnny-mnemonic/linux-ia64/commit/79a7beba4a685a7400ec075b1976bf49e7a8b458) (revert of the prior commit)

This reverts commit 2ec01bd7152f80795eab7b48706aa5db8d4a286a.
---
 include/asm-generic/vmlinux.lds.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/asm-generic/vmlinux.lds.h b/include/asm-generic/vmlinux.lds.h
index ae2d2359b79e..33d67f3c0a82 100644
--- a/include/asm-generic/vmlinux.lds.h
+++ b/include/asm-generic/vmlinux.lds.h
@@ -409,6 +409,7 @@ defined(CONFIG_AUTOFDO_CLANG) || defined(CONFIG_PROPELLER_CLANG)
 	__start_init_stack = .;						\
 	init_thread_union = .;						\
 	init_stack = .;							\
+	KEEP(*(.data..init_task))					\
 	KEEP(*(.data..init_thread_info))				\
 	. = __start_init_stack + THREAD_SIZE;				\
 	__end_init_stack = .;
-- 
2.25.1

